### PR TITLE
Implement constituents infrastructure in analytic IC routines

### LINKED ISSUE #287: Constituent variables aren't initialized properly when analytic ICs are enabled
### What happened?

When running with an `FKESSLER` compset and `ne30pg3` grid (which uses analytic ICs), it was found that certain constituent variables (specifically `Q` and `CLDLIQ`) are initialized to zero, which results in no moist processes (e.g. precipitation) occurring during  the moist baroclinic wave test.  This is not what is expected, and doesn't match how CAM is initializing the values.

### What are the steps to reproduce the bug?

Run the `FKESSLER` compset in CAM-SIMA and CAM with an `ne30pg3` grid with the zeroth (initial) time step outputted, and compare the `Q` and `CLDLIQ` values between the two different models.

### What CAM-SIMA hash were you using?

8ac3838

### What machine were you running CAM-SIMA on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

GNU

### Path to a case directory, if applicable

/glade/derecho/scratch/nusbaume/model_runs/SIMA_history_test

### Will you be addressing this bug yourself?

Yes

### Extra info

Technically this bug was found on the following CAM-SIMA branch:

https://github.com/peverwhee/CAM-SIMA/tree/history_court

But I have no reason to believe it doesn't exist on the main development branch as well (especially after PR #274 is eventually merged in).

### PR DESCRIPTION
**Pull request by nusbaume**
Originator(s): nusbaume, adamrher

Summary:

Adds in the subroutine/function hooks needed for the analytic ICs to properly interact with the CCPP constituents object.  This PR also contains some bug fixes for the air composition and SE dycore dynamics-physics coupling routines that were found via testing with the FKESSLER and FHS94 compsets.   Finally, this PR also includes some slight cleanup of the use of the water vapor constituent standard name within the core CAM-SIMA host code. 

Fixes #287

cod reviewed by:  adamrher, peverwhee

Describe any changes made to build system:  N/A

Describe any changes made to the namelist:

M       src/dynamics/tests/namelist_definition_analy_ic.xml
    - Added the US standard atmosphere option to the analytic IC types.

List any changes to the defaults for the input datasets (e.g. boundary datasets): N/A

List all files eliminated and why: N/A

List all files added and what they do: N/A

List all existing files that have been modified, and describe the changes:

M       src/control/cam_comp.F90
M       src/control/runtime_obj.F90
    - Added new "wv_stdname" parameter to cleanup water vapor standard name usage.

M       src/data/air_composition.F90
    -  Fix cpair bug when no dry air species are listed, and use new "wv_stdname" parameter.

M       src/dynamics/se/dp_coupling.F90
    -  Implement qneg and remaining constituents infrastructure, and add missing wet-to-dry conversion step.

M       src/dynamics/se/dycore/prim_state_mod.F90
    -  Whitespace cleanup.

M       src/dynamics/se/dyn_comp.F90
M       src/dynamics/tests/initial_conditions/ic_baro_dry_jw06.F90
M       src/dynamics/tests/initial_conditions/ic_baroclinic.F90
M       src/dynamics/tests/initial_conditions/ic_held_suarez.F90
M       src/dynamics/tests/initial_conditions/ic_us_standard_atm.F90
    - Implement new constituents infrastructure in analytic IC routines.

If there are new failures (compare to the existing-test-failures.txt file),
have them OK'd by the gatekeeper, note them here, and add them to the file.
If there are baseline differences, include the test and the reason for the
diff. What is the nature of the change? Roundoff?

derecho/intel/aux_sima:  All Pass

derecho/gnu/aux_sima:  All Pass

CAM-SIMA date used for the baseline comparison tests if different than latest:

### CODE REVIEW

File: src/dynamics/tests/initial_conditions/ic_baroclinic.F90 [hunk 1]
```diff
@@ -306,12 +315,14 @@ subroutine bc_wav_set_ic(vcoord,latvals, lonvals, zint, U, V, T, PS, PHIS, &
                 pk = moist_pressure_given_z(zk(k),latvals(i))
                 qk = qv_given_moist_pressure(pk,latvals(i))
               else
-                qk = 0.d0
+                qk = 0._r8
+              end if
+              if (lq) then
+                Q(i,k,m_cnst_ix_q) = qk
               end if
-              if (lq .and. cnst1_is_moisture) Q(i,k,1) = qk
```
Comment (adamrher) [inline src/dynamics/tests/initial_conditions/ic_baroclinic.F90:311]: Is the problem that Q was not getting set by ```bc_wav_set_ic```? While this is a clunkier way to set Q, shouldn't it still work? Would you mind explaining to me how this buggy code is resulting in Q incorrectly being zero everywhere?

**Review by adamrher — APPROVED** [review]
Looks good to me. I think I understand how you're using the new constituent infrastructure to get indices and get defaults constituent values in these ic_xxx.F90 modules. But I did have a question below about what exactly was wrong with the FKESSLER compset, and how this fixes it.

File: src/dynamics/se/dyn_comp.F90 [hunk 2]
```diff
@@ -9,7 +9,8 @@ module dyn_comp
 use cam_constituents,       only: const_get_index, const_is_wet, const_qmin
 use cam_constituents,       only: readtrace
 use air_composition,        only: const_is_water_species
-use cam_control_mod,        only: initial_run, simple_phys
+use cam_control_mod,        only: initial_run
+use cam_physics_control,    only: simple_phys
```
Comment (peverwhee) [inline src/dynamics/se/dyn_comp.F90:13]: i think this should come from `cam_control_mod` (as `cam_physics_control` won't come in until my history PR) 

::grimace::

File: src/dynamics/tests/initial_conditions/ic_baro_dry_jw06.F90 [hunk 3]
```diff
@@ -236,40 +243,63 @@ subroutine bc_dry_jw06_set_ic(vcoord, latvals, lonvals, U, V, T, PS, PHIS, &
...
+
+          if (iret /= 0) then
+            call endrun(errmsg, file=__FILE__, line=__LINE__)
+          end if
+
+          if (const_has_default) then
+
+            !If default value exists, then extract default value
+            !from constituent properties object:
+            call const_props(m_cnst(m))%default_value(const_default_value, &
```
Comment (peverwhee) [inline src/dynamics/tests/initial_conditions/ic_baro_dry_jw06.F90:285]: I think we're missing this line somewhere prior to this:

```
const_props => cam_model_const_properties()
```

**Review by peverwhee — APPROVED** [review]
two small things!

File: src/dynamics/se/dyn_comp.F90 (hunk 2 above)
Comment (nusbaume) [reply src/dynamics/se/dyn_comp.F90]: Good catch!  Definitely missed that line during cleanup.

File: src/dynamics/tests/initial_conditions/ic_baro_dry_jw06.F90 (hunk 3 above)
Comment (nusbaume) [reply src/dynamics/tests/initial_conditions/ic_baro_dry_jw06.F90:294]: Great catch!  In fact it looks like I am missing way more than that (I am kinda surprised my original tests ran at all...).

**Review by peverwhee — APPROVED** [review]
looks good to me!

**Pull request merged**